generator/internal/dart/templates/lib/service.mustache

```diff
@@ -17,8 +17,10 @@ limitations under the License.
 {{#Codec.DocLines}}
 {{{.}}}
 {{/Codec.DocLines}}
-class {{Codec.Name}} {
-  static const String defaultHost = 'https://{{DefaultHost}}';
+class {{Codec.Name}} extends CloudService {
```

Comment:
Q: are these `extend ...` something that application developers can depend on in their code?  Can the cast to the base type?  Probably fine either way, just trying to see the tradeoffs.

Reply:
Using inheritance here isn't strictly necessary. We could use composition and generate (modestly) more code in each service. The application developer likely won't ever cast to `CloudService` or make use of it directly.

Having messages and enums share a common type hierarchy will make the json encoding simpler. Using a common base class for services won't really be any better or worse than composition.

Reply:
> Having messages and enums share a common type hierarchy will make the json encoding simpler.

I can see how that would help.  Probably keep it that way then.

> Using a common base class for services won't really be any better or worse than composition.

I think we should go for composition then.  We can always revert that decision, but using a base class commits us to using a base class forever.

Reply:
> but using a base class commits us to using a base class forever

sgtm! I'll plan to make that change in the near term